doc/compiler/optimizer/LocalOptimizationsSummary.md

```diff
@@ -0,0 +1,103 @@
+the terms of the Eclipse Public License 2.0 which accompanies this
+distribution and is available at https://www.eclipse.org/legal/epl-2.0/
+or the Apache License, Version 2.0 which accompanies this distribution and
+is available at https://www.apache.org/licenses/LICENSE-2.0.
+
+This Source Code may also be made available under the following
+Secondary Licenses when the conditions for such availability set
+forth in the Eclipse Public License, v. 2.0 are satisfied: GNU
+General Public License, version 2 with the GNU Classpath
+Exception [1] and GNU General Public License, version 2 with the
+OpenJDK Assembly Exception [2].
+
+[1] https://www.gnu.org/software/classpath/license.html
+[2] http://openjdk.java.net/legal/assembly-exception.html
+
+SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+-->
+
+# Local Optimizations
+
+Local optimizations operate on extended basic blocks at a time.
+
+## Tree Simplification
+Tree Simplification performs simple IL tree transformations like constant
+folding, strength reduction, canonicalizing expressions (e.g. an `add` with a
+constant child will have the constant as the second child), branch/switch
+simplification etc. Float/double simplification support needs to be improved
+(call C libraries to compute the correct answer as per the Java spec).
+
+## Local Common Subexpression Elimination (CSE)/Copy Propagation
+Local CSE/Copy Propagation commons identical expressions within a block and
+performs removal of identical checks along with traditional commoning. Copy
+Propagation is done simultaneously (The right hand side (RHS) of a store is
+evaluated and the resulting register is used at each use of the stored value).
+Commoning is done on syntactic equivalence.
+
+## Local Value Propagation
+Local [Value Propagation](https://github.com/eclipse/omr/blob/master/doc/compiler/optimizer/ValuePropagation.md)
+performs constant, type and relational propagation within a block.
+It propagates constants and types based on assignments and performs removal
+of subsumed checks and compares. It also performs devirtualization
+of calls, checkcast and array store check elimination. It does not do value
+numbering, instead it works on the assumption that every expression has a
+unique value number (improved effectiveness if expressions having same value
+are commoned by a prior local CSE pass).
+
+## Local Dead Store Elimination
+Local Dead Store Elimination eliminates stores to locals/fields/statics (if
+there are two stores to the same memory without any intervening use in the
+same block).
+
+## Local Reordering of Expressions
+Local Reordering of Expressions reduces live ranges (register pressure) by
+moving evaluations around within a block.
+
+## Array Length Simplification
+In OpenJ9, array lengths are stored in number of bytes. So bound checks require
+a shift right (depending on the array stride) on the array length so that
```

Comment:
```suggestion
a shift right (depending on the array stride) on the array length so that the
```

Reply:
This optimization no longer is needed since arraylengths were changed in OpenJ9 to be stored as "number of elements" rather than "number of bytes" 20 years ago :)

Reply:
To be 100% clear, I am saying this optimization does not exist for the above reason anymore and so it also should not be documented.